dace/subsets.py

```diff
@@ -813,6 +813,17 @@ def intersects(self, other: 'Range'):
 
         return True
 
+    @staticmethod
+    def are_intersecting(ranges: List['Range']) -> bool:
+        """ Check if any of the given ranges are intersecting
+            :param ranges: the ranges to be checked.
+        """
+        for i in range(len(ranges)):
+            for j in range(i+1, len(ranges)):
+                if ranges[i].intersects(ranges[j]):
```

Comment:
Subset.intersects has four potential return values: True (if definitely intersects), False (if definitely does not intersect), None (unsupported cases), or raising TypeError (if the result is indeterminate).
Additionally, Range will not cover all cases (e.g., Indices).

You may want to use [dace.subsets.intersects](https://github.com/spcl/dace/blob/master/dace/subsets.py#L1294), which has only three return values (True, False, None) and accepts all subset types. Also, you will need to address the None case (ignoring it). Lastly, I would move the loop itself to the validation routine and not add another API for this.